official/transformer/v2/attention_layer.py

```diff
@@ -21,6 +21,14 @@
 import tensorflow as tf
 
 
+def _float32_softmax(logits, name=None):
+  """Computes a softmax activation in float32."""
+  input_dtype = logits.dtype
+  logits = tf.cast(logits, tf.float32)
+  output = tf.nn.softmax(logits, name=name)
```

Comment:
Would it be appropriate to add some documentation for why do we need to re-cast to fp32 right before softmax and then back?

Reply:
Done. Updated the docstring.